src/uproot/interpretation/library.py

```diff
@@ -840,105 +839,22 @@ def imported(self):
 
     def finalize(self, array, branch, interpretation, entry_start, entry_stop):
         pandas = self.imported
+        index = _pandas_basic_index(pandas, entry_start, entry_stop)
 
-        if isinstance(array, uproot.interpretation.objects.StridedObjectArray):
-            arrays = []
-            columns = []
-            _strided_to_pandas((), array.interpretation, array.array, arrays, columns)
-            maxlen = max(len(x) for x in columns)
-            if maxlen == 1:
-                columns = [x[0] for x in columns]
-            else:
-                columns = pandas.MultiIndex.from_tuples(
-                    [x + ("",) * (maxlen - len(x)) for x in columns]
-                )
-            index = _pandas_basic_index(pandas, entry_start, entry_stop)
-            out = pandas.DataFrame(
-                dict(zip(columns, arrays)), columns=columns, index=index
-            )
-            out.leaflist = maxlen != 1
-            return out
-
-        elif isinstance(array, uproot.interpretation.jagged.JaggedArray) and isinstance(
-            array.content, uproot.interpretation.objects.StridedObjectArray
+        if (
+            (not (isinstance(array, numpy.ndarray)))
+            or (array.dtype.names is not None)
+            or (len(array.shape) != 1)
         ):
-            index = pandas.MultiIndex.from_arrays(
-                array.parents_localindex(entry_start, entry_stop),
-                names=["entry", "subentry"],
-            )
-            arrays = []
-            columns = []
-            _strided_to_pandas(
-                (), array.content.interpretation, array.content.array, arrays, columns
+            awkward_pandas = uproot.extras.awkward_pandas()
+            ak_lib = Awkward()
```

Comment:
These objects should be treated as singletons.

```suggestion
            ak_lib = _libraries[Awkward.name]
```

Reply:
Done